src/components/AnimalList/AnimalList.jsx

```diff
@@ -0,0 +1,16 @@
+import React from "react";
+import Animal from "../Animal/Animal";
+
+const AnimalList = ({ animals, handleOnClick }) => {
```

Comment:
zase, cosmetic: 
- funkci: "handleOnClick" v kompoentě Animal bych nazval "onAnimalClick" - protože podobná logika/konvence jako onClick/onFocus/..., tj chybí tam to slovíčko handle. Komponenta poskytuje nějaký callback/event na "když se něco stane" (onXxxx).
- handleOnClick v AnimalList bych nazval onAnimalChange, a tj tady to měl jako
`onAnimalClick={(id) => onAnimalChange(id)}` (bez typescriptu bych to nezkracoval na `onAnimalClick={onAnimalChange}` protože se lze lehce splést). A to dle logiky že neklikáš na AnimalList ale AnimalList poskytuje event že se zvíře změnilo

To už jsem ale hodně hnidopich :D